### PR TITLE
Restore HP if it's low during copychain

### PR DESCRIPTION
**Pull request by Ignose**
People are DYING out there, DYING!!! - so let's make sure they heal.

### CODE REVIEW

**Comment by horrible-little-slime** [comment]
Not convinced on this for two reasons:
1. The condition seems wrong--shouldn't it be `&&`, not `||`?
2. We do restore hp between embezzler fights, when we run `postCombatActions` in fights.ts. If we're going to augment our healing, we should do so there.

File: garbage-collector
Comment (horrible-little-slime) [inline garbage-collector:1]: In addition to my previous review, please stop committing whatever this is

Comment (Rinn) [reply garbage-collector:1]: postCombatActions() is run after free fights in src\fights.ts and some other places but not during src\embezzlers\fights.ts and i don't think it even does hp recovery. maybe should be just calling safeRestore()?

**Comment by Rinn** [comment]
> 
>     2. We do restore hp between embezzler fights, when we run `postCombatActions` in fights.ts. If we're going to augment our healing, we should do so there.

postCombatActions() is run after free fights in src\fights.ts and some other places but not during src\embezzlers\fights.ts and i don't think it even does hp recovery. maybe should be just calling safeRestore()?

**Comment by horrible-little-slime** [comment]
> postCombatActions() is run after free fights in src\fights.ts and some other places but not during src\embezzlers\fights.ts and i don't think it even does hp recovery. maybe should be just calling safeRestore()?

```typescript
      // REMAINING EMBEZZLER FIGHTS
      let nextFight = getNextCopyTargetFight();
      while (nextFight !== null && myAdventures()) {
        print(`Running fight ${nextFight.name}`);
        const startTurns = totalTurnsPlayed();

        if (
          nextFight.draggable === "backup" &&
          have($skill`Musk of the Moose`) &&
          !have($effect`Musk of the Moose`)
        ) {
          useSkill($skill`Musk of the Moose`);
        }

        if (nextFight.gregariousReplace) {
          const crateIsSabered = get("_saberForceMonster") === $monster`crate`;
          const notEnoughCratesSabered = get("_saberForceMonsterCount") < 2;
          const weWantToSaberCrates = !crateIsSabered || notEnoughCratesSabered;
          if (weWantToSaberCrates) saberCrateIfSafe();
        }

        const location = new EmbezzlerFightRunOptions(nextFight).location;
        const underwater = location.environment === "underwater";

        const famSpec = familiarSpec(underwater, nextFight.name);

        setLocation(location);
        embezzlerOutfit({ ...nextFight.spec, ...famSpec }, location).dress();

        runEmbezzlerFight(nextFight, { action: nextFight.name });
        postCombatActions();

        print(`Finished ${nextFight.name}`);
        if (
          totalTurnsPlayed() - startTurns === 1 &&
          get("lastCopyableMonster") === globalOptions.target &&
          (nextFight.wrongEncounterName ||
            get("lastEncounter") === globalOptions.target.name)
        ) {
          eventLog.initialCopyTargetsFought++;
          eventLog.copyTargetSources.push(nextFight.name);
        }

        nextFight = getNextCopyTargetFight();

        if (
          romanticMonsterImpossible() &&
          (!nextFight || !nextFight.draggable)
        ) {
          doSausage();
          yachtzee();
        }
        doGhost();
        startWandererCounter();
      }
    });
    ```
it does get run in the embezzler parts of fights.ts!

Re-using `safeRestore` is a good idea, but again, I'd like to identify where the actual problem is, because we are running our post combat actions after these fights.

**Comment by Rinn** [comment]
bleah i wish i had time to convert this to grimoire tasks it's too complicated right now

**Pull request closed without merging**